### PR TITLE
Refactor Request Handler

### PR DESCRIPTION
**Pull request by jubalm**

### CODE REVIEW

File: app/ts/simulation/services/EthereumJSONRpcRequestHandler.ts [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
 		this.caching = caching
 		this.cache = new Map()
 		this.pendingCache = new Map()
-    }
-	public readonly getRpcEntry = () => this.rpcEntry
+		this.getRpcEntryFromStorage()
+	}
+
+
+	private getRpcEntryFromStorage = async () => {
```
Comment (KillariDev) [inline app/ts/simulation/services/EthereumJSONRpcRequestHandler.ts:30]: I don't think we should retrieve the entry from the storage. This class doesn't need the entry and it should work without it

Comment (KillariDev) [reply app/ts/simulation/services/EthereumJSONRpcRequestHandler.ts:30]: ah sorry. This class needs it, and it should take it as input. I don't think the API of this class should change

File: app/ts/simulation/services/EthereumJSONRpcRequestHandler.ts [hunk 2]
```diff
@@ -59,14 +58,19 @@ export class EthereumJSONRpcRequestHandler {
 		}
 	}
 
+	public getChainId = async () => {
+		const response = await this.jsonRpcRequest({ method: 'eth_chainId', params: [] })
+		return funtypes.BigInt.parse(response)
```
Comment (KillariDev) [inline app/ts/simulation/services/EthereumJSONRpcRequestHandler.ts:63]:
```suggestion
		return EthereumQuantity.parse(response)
```

File: app/ts/simulation/services/EthereumClientService.ts [hunk 3]
```diff
@@ -18,15 +19,17 @@ export class EthereumClientService {
...
 	private onErrorBlockCallback: (ethereumClientService: EthereumClientService) => Promise<void>
 	private requestHandler
+	private rpcEntry
 
-    constructor(requestHandler: IEthereumJSONRpcRequestHandler, newBlockAttemptCallback: (blockHeader: EthereumBlockHeader, ethereumClientService: EthereumClientService, isNewBlock: boolean) => Promise<void>, onErrorBlockCallback: (ethereumClientService: EthereumClientService) => Promise<void>) {
+    constructor(requestHandler: IEthereumJSONRpcRequestHandler, newBlockAttemptCallback: (blockHeader: EthereumBlockHeader, ethereumClientService: EthereumClientService, isNewBlock: boolean) => Promise<void>, onErrorBlockCallback: (ethereumClientService: EthereumClientService) => Promise<void>, rpcEntry: RpcEntry) {
 		this.requestHandler = requestHandler
 		this.newBlockAttemptCallback = newBlockAttemptCallback
 		this.onErrorBlockCallback = onErrorBlockCallback
+		this.rpcEntry = rpcEntry
```
Comment (KillariDev) [inline app/ts/simulation/services/EthereumClientService.ts:28]: add a check here that request handlers url matches to the rpcentries url, if not, throw an error

File: app/ts/types/JsonRpc-types.ts [hunk 4]
```diff
@@ -402,6 +408,7 @@ export const EthereumJsonRpcRequest = funtypes.Union(
 	UninstallFilter,
 	GetFilterChanges,
 	GetFilterLogs,
+	EthChainId,
```
Comment (KillariDev) [inline app/ts/types/JsonRpc-types.ts:411]: this aleady existst above:
`funtypes.ReadonlyObject({ method: funtypes.Literal('eth_chainId') }),`, remove that if you need to have it separated.

File: app/ts/types/JsonRpc-types.ts [hunk 5]
```diff
@@ -357,6 +357,12 @@ export const GetFilterLogs = funtypes.ReadonlyObject({
 	params: funtypes.ReadonlyTuple(funtypes.String)
 })
 
+export type EthChainId = funtypes.Static<typeof EthChainId>
+export const EthChainId = funtypes.ReadonlyObject({
+	method: funtypes.Literal('eth_chainId'),
+	params: funtypes.ReadonlyArray(funtypes.Never),
```
Comment (KillariDev) [inline app/ts/types/JsonRpc-types.ts:363]: params field is not mandatory for `eth_chainId`

File: test/MockRequestHandler.ts [hunk 6]
```diff
@@ -7,15 +7,6 @@ export class MockRequestHandler {
...
-		chainId: 5n,
-		httpsRpc: 'https://rpc-goerli.dark.florist/flipcardtrustone',
-		currencyName: 'Goerli Testnet ETH',
-		currencyTicker: 'GÖETH',
-		primary: true,
-		minimized: true,
-		weth: 0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6n,
-	})
+
+	public getChainId = async () => 0n
```
Comment (KillariDev) [inline test/MockRequestHandler.ts:11]:
```suggestion
	public getChainId = async () => 5n
```

File: test/tests/nethermindComparison.ts [hunk 7]
```diff
@@ -17,13 +16,10 @@ function parseRequest(data: string) {
...
 
 class MockEthereumJSONRpcRequestHandler {
-	private rpcEntry: RpcEntry
-	constructor(rpcEntry: RpcEntry, _caching = false) {
-		this.rpcEntry = rpcEntry
-	}
-
 	public clearCache = () => {}
 
+	public getChainId = async () => 0n
```
Comment (KillariDev) [inline test/tests/nethermindComparison.ts:21]:
```suggestion
	public getChainId = async () => 5n
```

File: app/ts/simulation/services/EthereumClientService.ts (hunk 3 above)
Comment (jubalm) [reply app/ts/simulation/services/EthereumClientService.ts:28]: added in [f0232a5](https://github.com/DarkFlorist/TheInterceptor/pull/1038/commits/f0232a50f52630cb66b9d6670b86ed1db7bb91c3)

File: app/ts/types/JsonRpc-types.ts (hunk 5 above)
Comment (jubalm) [reply app/ts/types/JsonRpc-types.ts]: removed as the inline definition will already suffice (missed that one out)

**Pull request merged**